### PR TITLE
Feature/breadcrumb functionality

### PR DESCRIPTION
**Pull request by maazahmed12**
Created Breadcrumbs. 

Some shortcomings are mentioned below:
The breadcrumb feature will not work on the pages whose components are not created, e.g (few pages under my-account tab).
 For products and categories (excluding brands) their names are not displayed because of the following reason:
       1) Product name is not available in route url.
       2) Category title is present in route url but more specific code is required.

### CODE REVIEW

**Comment by kacperknapik** [comment]
Hi @maazahmed12! I'm currently taking care of #136 and it seems your work could help me. But first things first - there is a huge possibility that you've created your branch, starting from `master` instead of `develop`. That's why your PR consists of so many commits. Please fix that before we can go any further.

File: package.json [hunk 1]
```diff
@@ -105,4 +105,4 @@
     "typescript": "~3.1.6",
     "webpack-cli": "^3.1.2"
   }
-}
+}
```
Comment (kacperknapik) [inline package.json]: according to our prettier config, this newline should stay here

File: projects/storefrontapp/src/environments/environment.ts [hunk 2]
```diff
@@ -5,5 +5,5 @@
 
 export const environment = {
   production: false,
-  occBaseUrl: ''
+  occBaseUrl: 'https://localhost:9002'
```
Comment (kacperknapik) [inline projects/storefrontapp/src/environments/environment.ts]: you shouldn't commit this file. please leave it as it was - empty string

**Comment by Xymmer** [comment]
Hey @maazahmed12, we started looking into implementing your original breadcrumbs PR but because of changes in routing and how we work with the backend, we couldn't use your code. Then investigating we see it's a bit more complicated then we thought. 

Currently we're spiking this issue - see  #136.

**Comment by maazahmed12** [comment]
Hey @Xymmer, Thanks for looking into my PR. I would like to know why the code is complicated and I am keen to resolve its complexity. Can I also contribute in the spike so we can really make use of my code.

**Pull request closed without merging**